src/displayBackend/wayland/Surface.hpp

```diff
@@ -61,7 +61,7 @@ class Surface
 
 	const uint32_t cFrameTimeoutMs = 50;
 
-	Surface(wl_compositor* compositor);
+	explicit Surface(wl_compositor* compositor);
```

Comment:
> Your commit description is misleading.
> 
> `assert(compositor != nullptr)` will lead to crash as well. It will be controllable crash, but anyways...
> 
> Also, s/checke/checked.

I afford to disagree
1.Why assert - because we do not have a deal with he run-time binding, all types and parameters are known at compilation time and no sense to do this verification during runtime. So, this problem during the 1st attempt of the object initialization. The asset helps to understand the line of code where crash happens.
2.Review https://github.com/xen-troops/displ_be/pull/122, where it is proposed to use ASSERT in the similar situation.

Reply:
Phrasing of your commit description  suggests that this commit will avoid the crash.
Instead, it will make crash controllable, by using this `assert`. This is a right thing.

But commit message should be reworded.